docs/tutorial/asar-archives.md

```diff
@@ -0,0 +1,175 @@
+---
+title: ASAR Archives
+description: What is ASAR archive and how does it affect the application.
+slug: asar-archives
+hide_title: false
+---
+
+After creating an [application distribution](application-distribution.md), the
+app's source code are usually bundled into an [ASAR
+archive](https://github.com/electron/asar), which is a simple extensive archive
+format designed for Electron apps. By bundling the app we can mitigate issues
```

Comment:
The word "issue" was originally linked to https://github.com/nodejs/node-v0.x-archive/issues/6960 . But hasn't this been fixed years ago (npm doesn't nest `node_modules` any longer)?

Electron also recommends bundling your code (https://www.electronjs.org/de/docs/latest/tutorial/performance#7-bundle-your-code), which also speeds up `require` (they're mostly gone) and "conceals" the source (by means of minification and mangling).

The reason why I'm bringing this up is that when I started with Electron I read this paragraph and couldn't find a single compelling reason to use ASAR (and didn't), only problems (e.g. the `Worker` constructor didn't seem to be patched to add ASAR support).

Is there actually a real benefit of using ASAR? I'm curious if I'm missing out.

Summary:

1. Long path names haven't been an issue since 2015 (?)
2. There's no evidence linked that ASAR is faster than bundling (which is recommended anyway), maybe include a benchmark somewhere?
3. "conceal your source code from cursory inspection" is a really weak argument, if at all

So maybe this could be rewritten for 2022?

Reply:
Bundlers are not good at handling assets as far as I know. For example you can not bundle image files and then serve them via `protocol` module, or execute a bundled python script with `child_process`. Of course you can work around them if your app is designed to work with bundlers from start, but ASAR is designed to work with arbitrary apps and package them into one single file.

But still I agree ASAR does not provide a real benefit than bundling, and this documentation means to list the limitations of ASAR instead of promoting it.

Node.js is going to work on a similar thing https://github.com/nodejs/single-executable, and we might want to use it as the default format in the end.

Reply:
Thanks for clarifying. It sounds like you brought up some good points that should belong into the docs (maybe we could explicitly mention bundling and the pros and cons).

> you can not bundle image files and then serve them via `protocol` module

For me they are bundled as part of the _client_ bundle and I point `@fastify/static` (which is bundled in the server bundle) to the `client/dist` folder. But I understand not everyone is running an HTTP server. This works as expected for me:

```
import chromeProxySettings from '~/images/getting-started-chrome-proxy.png';
<img src={chromeProxySettings} alt="Chrome proxy settings" />
```
I don't see why the `protocol` module couldn't serve files from my `client/dist` folder as well.

> or execute a bundled python script with `child_process`

I think ASAR makes that _harder_, not easier? Because depending on the circumstances you'd have to explicitly do something like `asarUnpack` (for electron-builder) for it to even work. I'm shipping a Python binary (via PyInstaller) with additional Python scripts and I don't have any problems so far using `child_process.spawn`.

Reply:
Yeah this document needs an update, and actually the original PR that removed it was planning to move the document to [the asar repo](https://github.com/electron/asar) but somehow the effort ended nowhere.

This PR means to bring back the deleted page since it contains important information that currently can not be found on Internet, we might delete it later after getting a renewed documentation in the asar repo.